src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```
@@ -75,7 +75,7 @@ public class ActivityTagsCollection : System.Collections.Generic.IDictionary<str
     {
         public ActivityTagsCollection() { throw null; }
         public ActivityTagsCollection(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object>> list) { throw null; }
-        public object? this[string key] { get { throw null; } set { } }
+        public object this[string key] { get { throw null; } set { } }
```

This might need to be nullable, didn't understood why made non null, but if I change it to nullable the entire collection nullability need to be changed

---

Just taking a quick look at the implementation, it looks like this should be `object?` with `DisallowNull`, and the implementation should be changed to remove the `!` from the `return index < 0 ? null! : _list[index].Value;`.  I'm not sure why it was suppressed like that.
cc: @tarekgh 
(and if it was done there, seems like the rest of the code should be quickly audited to make sure the same suppression wasn't used elsewhere in similar cases)

---

missed this feedback

---

@stephentoub it seems this is implementation of `IDictionary<string, object>` if I changed the nullability of the base there will be conflict between DisallowNull and base. IMO we should change nullability to `object?` but to not annotate this as `DisallowNull` and just let it throw or add the attribute and suppress the warning... suggestions?

---

> Just taking a quick look at the implementation, it looks like this should be object? with DisallowNull

The indexer allow nulls so the original fix should be ok. 
```C#
 public object this[string key] { get { throw null; } set { } }
```

basically the indexer allow null in the getter but never return null.

---

looks I confused myself. I think @stephentoub was right. it should be `object?`. so you may ignore my previous comment.

---

I am now wondering what `DisallowNull` will do? does it will prevent the setter from using null values? of so then this will be wrong.

---

Yes, it will warn if somebody set null value, from reading the API comments again, it should allow null, also it returns null when `index < 0`: `return index < 0 ? null! : _list[index].Value;`. So it need to be just nullable. That could affect the entire dictionary: to allow Add null and return nullable Values etc which seems not intended. I guess we just make it nullable and suppress the warning for the generic parent.

---

for sure the indexer should allow null so it should be nullable. DisallowNull should be removed.